app/ytdl.py

```diff
@@ -210,8 +210,15 @@ async def __add_entry(self, entry, quality, format, already):
         if etype == 'playlist':
             entries = entry['entries']
             log.info(f'playlist detected with {len(entries)} entries')
+            playlist_index_digits = len(str(len(entries)))
             results = []
-            for etr in entries:
+            for index, etr in enumerate(entries, start=1):
+                etr["playlist"] = entry["id"]
+                etr["playlist_index"] = '{{0:0{0:d}d}}'.format(playlist_index_digits).format(index)
+                etr["playlist_id"] = entry["id"]
+                etr["playlist_title"] = entry["title"]
+                etr["playlist_uploader"] = entry["uploader"]
+                etr["playlist_uploader_id"] = entry["uploader_id"]

```

Comment:
can we be certain that the "uploader" and "uploader_id" properties will always exist in those entries? yt-dlp supports a lot of sites, maybe the extractors for some of them don't fill those fields... perhaps something like this can be done:
```suggestion
                for property in ("id", "title", "uploader", "uploader_id"):
                    if property in entry:
                        etr[f"playlist_{property}"] = entry[property]
```